src/core/gui/XournalView.cpp

```diff
@@ -717,10 +726,14 @@ void XournalView::documentChanged(DocumentChangeType type) {
     for (auto&& page: viewPages) { delete page; }
     viewPages.clear();
 
-    this->cache->clearCache();
+    this->cache.reset();
 
     Document* doc = control->getDocument();
     doc->lock();
+    if (doc->getPdfPageCount() != 0) {
+        this->cache = std::make_unique<PdfCache>(doc->getPdfDocument(), control->getSettings()->getPdfPageCacheSize());
+        this->cache->setRefreshThreshold(control->getSettings()->getPDFPageRerenderThreshold());
+    }
```

Comment:
Please refactor this (and the above) into a (private) helper function in case we add another setting to PdfCache in the future.